src/js/common/components/LoginModal.jsx

```diff
@@ -11,17 +11,35 @@ class LoginModal extends React.Component {
   constructor(props) {
     super(props);
     this.handleLogin = this.handleLogin.bind(this);
+
+    // I don't like this, but we need to make sure componentWillReceiveProps
+    //  doesn't call onLogin() when a page with a LoginModal is refreshed and
+    //  a user is logged in.
+    this.state = {
```

Comment:
I mentioned this below, but since it doesn't look like this is used in rendering, I think you can make it an instance property: `this.loginButtonClicked = false`.